wfl/calculators/espresso.py

```diff
@@ -123,54 +126,46 @@ def calculate(self, atoms=None, properties=_default_properties, system_changes=a
             self.parameters = deepcopy(self.initial_parameters)
 
     def setup_calc_params(self, properties):
-
-        # first determine if we do a non-periodic calculation.
-        # and update the properties that we will use.
-        nonperiodic, properties = handle_nonperiodic(self.atoms, properties, allow_mixed=True)
-
+        """Setup calculator params based on atoms structure (pbc) and requested properties
+        """
         # update the parameters with the cool wfl logic
         self.parameters["tprnfor"] = "forces" in properties
         self.parameters["tstress"] = "stress" in properties
 
-        if nonperiodic:
-            if not np.any(self.atoms.get_pbc()):
-                # FFF -> gamma point only
-                self.parameters["kpts"] = None
-                self.parameters["kspacing"] = None
-                self.parameters["koffset"] = False
+        if np.all(~self.atoms.pbc):
+            # FFF -> gamma point only
+            self.parameters["kpts"] = None
+            self.parameters["kspacing"] = None
+            self.parameters["koffset"] = False
+        elif np.any(~self.atoms.pbc):
+            # mixed T & F
+            if "kspacing" in self.parameters:
+                # need to create the grid,
+                # `kspacing` overwrites `kpts`
+                # we set it in there and
+                self.parameters["kpts"] = kspacing_to_grid(
+                    self.atoms, spacing=self.parameters["kspacing"] / (2.0 * np.pi)
```

Comment:
@bernstei Do we want to stick with the `ase.io.espresso kspacing_to_grid` function in the qe calculator or do we switch to the `universal_kspacing_n_k`? I think switching it could be good to make the code more uniform. However as far as I've seen the codes both function have basically the same function.

Reply:
I think we should use the same function for all the calculators, so it's actually universal, but I'm ambivalent about doing our own vs. using ASE's (the usual tradeoff of control vs. not doing extra work). If they give the same results, we can use the ASE for now, and switch back to our own implementation if we need something more, or if they break it.